### PR TITLE
Defer loading files until after layout restore

### LINKED ISSUE #3572: Defer loading file contents until after App is restored
This avoids hitting the server with potentially large uploads during the startup phase.  This will have to be done as part of #3495.

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #3572.  

Summary of changes:

API:
- Adds a new `initialize()` function to `Context` (but not `IContext`) that is used instead of the previous `save()` and `fromStore()` (which was only on `Context`).
- Adds an optional parameter to the document manager for `when` to load contents.

Behavioral:
- We no longer wait for the kernel to be ready before showing the file
- We no longer wait for the widget to be ready before returning from the `docmananger:open` command

### CODE REVIEW

File: packages/docregistry/src/context.ts [hunk 1]
```diff
@@ -208,80 +208,46 @@ class Context<T extends DocumentRegistry.IModel> implements DocumentRegistry.ICo
...
+   * Initialize the context.
    *
-   * @returns a promise that resolves upon model population.
+   * @param isNew - Whether it is a new file.
+   *
+   * @returns a promise that resolves upon initialization.
    */
-  fromStore(): Promise<void> {
+  initialize(isNew: boolean): Promise<void> {
+    if (isNew) {
```
Comment (ian-r-rose) [inline packages/docregistry/src/context.ts]: I just verified that this change still works with the `modelDB` in the Google Realtime stuff. I know that's going away soon, but wanted to make sure that we don't do it accidentally.

File: packages/docregistry/src/context.ts [hunk 2]
```diff
@@ -496,13 +425,109 @@ class Context<T extends DocumentRegistry.IModel> implements DocumentRegistry.ICo
         name,
         language: this._model.defaultKernelLanguage,
       };
-      return this.session.initialize();
+      this.session.initialize();
```
Comment (ian-r-rose) [inline packages/docregistry/src/context.ts]: I wonder if we shouldn't even wait on a checkpoint to resolve the ready promise. Thoughts?

Comment (blink1073) [reply packages/docregistry/src/context.ts]: Works for me, updated.

File: Untitled.ipynb [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{
```
Comment (ian-r-rose) [inline Untitled.ipynb]: Did you intend to add this file?

**Comment by ian-r-rose** [comment]
Thanks! This is a great improvement.

**Pull request merged**